crates/typst-pdf/src/color.rs

```diff
@@ -377,26 +377,37 @@ impl PaintEncode for Color {
 /// Extra color space functions.
 pub(super) trait ColorSpaceExt {
     /// Returns the range of the color space.
-    fn range(self) -> [f32; 6];
+    fn range(self) -> [f32; 8];
 
     /// Converts a color to the color space.
-    fn convert<U: QuantizedColor>(self, color: Color) -> [U; 3];
+    fn convert<U: QuantizedColor>(self, color: Color) -> Vec<U>;
 }
 
 impl ColorSpaceExt for ColorSpace {
-    fn range(self) -> [f32; 6] {
-        [0.0, 1.0, 0.0, 1.0, 0.0, 1.0]
+    fn range(self) -> [f32; 8] {
+        [0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0] // Not sure about that
```

Comment:
This is correct, but I would prefer if the range was dependent on the color space, for example `fn range(self)` could return `&'static [f32]` instead of a fixed size array, that could already be an improvement as we don't want to waste space.

Reply:
Tentative solution with `SmallVec`s (not sure what the impact on performance is, though): 

https://github.com/typst/typst/pull/4568/files#diff-55650bf2dd7b33ec02c37fd3b314eb3a5db0c2aa6bf0baf269ef9c03fe716c95L387-R394

```rust
    fn range(self) -> SmallVec<[f32; 8]> {
        match self {
            ColorSpace::Cmyk => smallvec![0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0],
            ColorSpace::D65Gray => smallvec![0.0, 1.0],
            _ => smallvec![0.0, 1.0, 0.0, 1.0, 0.0, 1.0],
        }
    }
```